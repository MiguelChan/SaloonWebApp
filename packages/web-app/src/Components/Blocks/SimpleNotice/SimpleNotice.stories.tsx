import React from 'react';
import {
  Meta,
  Story,
} from '@storybook/react';
import { SimpleNotice, SimpleNoticeProps } from '.';

export default {
  title: 'Components/Blocks/SimpleNotice',
  component: SimpleNotice,
} as Meta;

const Template: Story<SimpleNoticeProps> = (args) => <SimpleNotice {...args} />;

export const Default = Template.bind({});
Default.args = {
  noticeBody: 'This is the Notice Body',
  noticeTitle: 'This is the Notice Title',
};