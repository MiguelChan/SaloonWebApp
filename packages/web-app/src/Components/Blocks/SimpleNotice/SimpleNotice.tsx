import { Card, CardContent, Typography } from '@mui/material';
import React from 'react';

export interface SimpleNoticeProps {
  noticeTitle: string;
  noticeBody: string;
}

/**
 * Defines a Simple Notice to be displayed in the application.
 * 
 * @param props The properties of the SimpleNotice. 
 * @returns The SimpleNotice.
 */
export const SimpleNotice: React.FunctionComponent<SimpleNoticeProps> = ({
  noticeBody,
  noticeTitle,
}) => {
  return (
    <Card>
      <CardContent>
        <Typography sx={{fontSize: 14}} color='text.secondary' gutterBottom>
          {noticeTitle}
        </Typography>
        <Typography variant='h5' component='div'>
          {noticeBody}
        </Typography>
      </CardContent>
    </Card>
  );
};