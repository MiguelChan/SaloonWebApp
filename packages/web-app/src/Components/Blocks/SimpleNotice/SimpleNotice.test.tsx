import { 
  render, 
  RenderResult,
  screen,
} from '@testing-library/react';
import React from 'react';
import { SimpleNotice, SimpleNoticeProps } from './SimpleNotice';

describe('SimpleNotice', () => {

  it('Sohuld display the Notice', () => {
    const props: SimpleNoticeProps = {
      noticeBody: 'This is the Notice Body',
      noticeTitle: 'This is the Notice Title',
    };

    setupComponent(props);

    expect(screen.getByText(props.noticeBody)).toBeInTheDocument();
    expect(screen.getByText(props.noticeTitle)).toBeInTheDocument();
  });

  it('Should match the snapshot', () => {
    const props: SimpleNoticeProps = {
      noticeBody: 'This is the Notice Body',
      noticeTitle: 'This is the Notice Title',
    };

    const renderResult = setupComponent(props);

    expect(renderResult.container).toMatchSnapshot();
  });
;
  const setupComponent = (props: SimpleNoticeProps): RenderResult => {
    return render(<SimpleNotice {...props} />);
  }
});