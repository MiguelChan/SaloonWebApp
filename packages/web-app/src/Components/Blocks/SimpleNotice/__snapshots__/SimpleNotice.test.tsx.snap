// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleNotice Should match the snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-16m3jxf-MuiTypography-root"
      >
        This is the Notice Title
      </p>
      <div
        class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
      >
        This is the Notice Body
      </div>
    </div>
  </div>
</div>
`;
